python/Ganga/GPIDev/Lib/Job/Job.py

```diff
@@ -1472,6 +1471,9 @@ def submit(self, keep_going=None, keep_on_fail=None, prepare=False):
             # split into subjobs
             rjobs = self._doSplitting()
 
+            if not rjobs or len(rjobs) == 1 and rjobs[0] is self:
```

Comment:
Yes, as per the code in `_doSplitting` it may return `[self]` when there are no subjobs which means that no splitting occured but we have a splitter.